.github/workflows/reusable-release.yml

```diff
@@ -0,0 +1,839 @@
+                    , toolRequirements: "${{ needs.tool-output.outputs.rpm_tools }}"
+                    , DISTRO: "Fedora"
+                    , ARTIFACT: "x86_64-linux-fedora33"
+                    , ADD_CABAL_ARGS: "--enable-split-sections"
+                    },
+                    { image: "fedora:36"
+                    , installCmd: "dnf install -y"
+                    , toolRequirements: "${{ needs.tool-output.outputs.rpm_tools }}"
+                    , DISTRO: "Fedora"
+                    , ARTIFACT: "x86_64-linux-fedora36"
+                    , ADD_CABAL_ARGS: "--enable-split-sections"
+                    },
+                    { image: "fedora:38"
+                    , installCmd: "dnf install -y"
+                    , toolRequirements: "${{ needs.tool-output.outputs.rpm_tools }}"
+                    , DISTRO: "Fedora"
+                    , ARTIFACT: "x86_64-linux-fedora38"
+                    , ADD_CABAL_ARGS: "--enable-split-sections"
+                    },
+                    { image: "rockylinux:8"
+                    , installCmd: "yum -y install epel-release && yum install -y --allowerasing"
+                    , toolRequirements: "${{ needs.tool-output.outputs.rpm_tools }}"
+                    , DISTRO: "Unknown"
+                    , ARTIFACT: "x86_64-linux-rocky8"
+                    , ADD_CABAL_ARGS: "--enable-split-sections"
+                    },
+                    { image: "alpine:3.20"
+                    , installCmd: "apk update && apk add"
+                    , toolRequirements: "${{ needs.tool-output.outputs.apk_tools }}"
+                    , DISTRO: "Unknown"
+                    , ARTIFACT: "x86_64-linux-unknown"
+                    , ADD_CABAL_ARGS: "--enable-split-sections --enable-executable-static"
+                    },
+                    { image: "alpine:3.12"
+                    , installCmd: "apk update && apk add"
+                    , toolRequirements: "${{ needs.tool-output.outputs.apk_tools }}"
+                    , DISTRO: "Unknown"
+                    , ARTIFACT: "x86_64-linux-alpine312"
+                    , ADD_CABAL_ARGS: "--enable-split-sections"
+                    },
+                    { image: "alpine:3.20"
+                    , installCmd: "apk update && apk add"
+                    , toolRequirements: "${{ needs.tool-output.outputs.apk_tools }}"
+                    , DISTRO: "Unknown"
+                    , ARTIFACT: "x86_64-linux-alpine320"
+                    , ADD_CABAL_ARGS: "--enable-split-sections"
+                    }
+                  ]
+    container:
+      image: ${{ matrix.platform.image }}
+    steps:
+      - name: Install requirements
+        shell: sh
+        run: |
+          ${{ matrix.platform.installCmd }} curl bash git ${{ matrix.platform.toolRequirements }}
+
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ matrix.branch }}
+
+      - name: git clone fix
```

Comment:
Deserves a comment explaining what issue this is fixing.

Reply:
To my surprise, it appears cabal testsuite is running `git` to find the test files. Which is a bad hack. That causes problems in CI.

Validate pipeline does something similar:

https://github.com/haskell/cabal/blob/6d6fc4c772e1b8d3741d63878b8b2a35700fea08/.github/workflows/validate.yml#L321-L323